### DISCUSSION TITLE
Sub process in BPMN Editor

### DISCUSSION THREAD

**Discussion by NidhiMehata** [clientIO/joint#1752] [Q&A]
What is the right way to create a expanded sub process in the BPMN Editor. I don't see any native shape for the same

Demo URL: https://resources.jointjs.com/demos/bpmn
![image](https://user-images.githubusercontent.com/35390217/181468134-d4e895ae-1d7d-4b83-b76b-c5e63828a3bb.png)

PS,
The BPMN Activity has some parameters, but that is unclear.
![image](https://user-images.githubusercontent.com/35390217/181468309-7b6bc54f-cfde-4134-9610-6e7b6208bc68.png)

**Comment by kumilingus** [comment]
The BPMN Editor demo does not support defining sub-processes. The expanded sub-process is an activity (https://resources.jointjs.com/docs/rappid/v3.5/shapes.bpmn2.html#Activity).

You can:
1. open a new diagram when clicking on the activity: https://resources.jointjs.com/demos/tabs
2. use the embedding feature and collapse/expand the activity: https://resources.jointjs.com/demos/container

**Reply by NidhiMehata** [reply]
>  The expanded sub-process is an activity

But this does not act as a container, the shapes like activity/gateway cannot be placed inside it

> open a new diagram when clicking on the activity
How do we embed the tab links in the json, the demo does not seem to have code

> use the embedding feature and collapse/expand the activity:
Containers are not bpmn shapes, would they be rendered in bpmn editor

Why doesn't the BPMN Editor have a subprocess shape directly? What does the is subprocess option mean for a activity
![image](https://user-images.githubusercontent.com/35390217/181908647-f876ccea-448a-42bf-b670-c65fb81b1d5b.png)

**Reply by kumilingus** [reply]
> But this does not act as a container, the shapes like activity/gateway cannot be placed inside it

Any element in JointJS can become a container. Whether an element can become a child of another element is checked by the [validateEmbedding](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#dia.Paper.prototype.options.validateEmbedding) callback.

> Containers are not bpmn shapes, would they be rendered in bpmn editor

You can combine different shapes (e.g. BPMN, UML, DEVS, or custom) in a single application.

> How do we embed the tab links in the json, the demo does not seem to have code

The link to the source code can be found below the demo or [here](https://github.com/clientIO/joint/discussions/1666).

> Why doesn't the BPMN Editor have a subprocess shape directly?

It's a demo. It shows only limited features.

> What does the is subprocess option mean for a activity

I think it meant the type of Activity *Event Sub-Process* from [BPMN 2.0 specification](http://www.bpmb.de/images/BPMN2_0_Poster_EN.pdf).
